### PR TITLE
Add rgw xml unit tests

### PR DESCRIPTION
**Pull request by yuvalif**
This PR is mainly for adding unit-test for the RGW XML code.
However, as part of the PR:
- Documentation was added to rgw_xml.h, so that developers that need to do XML parsing will understand better how to use it
- The public API was minimized, so, that it is easier to understand how to use the library
All the changes were done in a way that existing code using the XML library does not need to be changed. This included:
  - Removing unused public API
  - Making public functions private/protected
  - Adding constness when possible

### CODE REVIEW

File: src/test/rgw/CMakeLists.txt [hunk 1]
```diff
@@ -151,3 +151,16 @@ if(WITH_RADOSGW_AMQP_ENDPOINT)
...
 endif()
+
+# unittest_rgw_xml
+add_executable(unittest_rgw_xml test_rgw_xml.cc)
+add_ceph_unittest(unittest_rgw_xml)
+
+target_link_libraries(unittest_rgw_xml rgw_a ${EXPAT_LIBRARIES})
+
+# unittest_rgw_xml
+add_executable(unittest_rgw_xml test_rgw_xml.cc)
```
Comment (cbodley) [inline src/test/rgw/CMakeLists.txt]: this `unittest_rgw_xml` is showing up twice in a row. it looks like your branch has two of the same commit somehow?

Comment (yuvalif) [reply src/test/rgw/CMakeLists.txt]: not sure how it happened (learning how to use git is an endless journey...), but hopefully fixed now.

**Comment by yuvalif** [comment]
Teuthology run:
http://pulpito.front.sepia.ceph.com/yuvalif-2019-02-27_21:25:22-rgw:verify-wip-yuval-add_rgw_xml_unit_tests-distro-basic-smithi/
All failures are valgrind issues related to mgr (nothing to do with RGW XML code).

**Comment by yuvalif** [comment]
regarding ```make check (arm64)``` failure - 167 of 168 tests passed and than the machine died...

**Pull request merged**